ocs_ci/ocs/platform_nodes.py

```diff
@@ -818,6 +824,271 @@ def _prepare_node(self, node_id):
             except Exception:
                 logger.error("Failed to create RHEL node")
                 raise
+        elif self.node_type == 'RHCOS':
+            conf = self._prepare_rhcos_node_conf()
+            conf['node_id'] = node_id
+            try:
+                self.aws_instance_obj = self._prepare_upi_rhcos_node(conf)
+            except Exception:
+                logger.error("Failed to create RHCOS node")
+                raise
+            self.approve_pending_csr()
+
+    def _prepare_rhcos_node_conf(self):
+        """
+        Merge default RHCOS node configuration for rhcos node
+        along with the user provided config
+
+        Returns:
+            conf (dict): A dictionary of merged user and default values
+
+        """
+        conf = self.read_default_config(constants.RHCOS_WORKER_CONF)
+        default_conf = conf.get('ENV_DATA')
+        merge_dict(default_conf, self.node_conf)
+        logger.info(f"Config after merge is {default_conf}")
+        return default_conf
+
+    def _prepare_upi_rhcos_node(self, conf):
+        """
+        Handle RHCOS worker instance creation using cloudformation template,
+        Create RHCOS instance with ami same as master
+
+        Args:
+            conf (dict): configuration for node
+
+        Returns:
+           awsinstance object
```

Comment:
Is there a class / type we can specify here?

Reply:
boto3.Instance

Reply:
done